src/Magick.NET/Formats/Png/PngWriteDefines.cs

```diff
@@ -0,0 +1,111 @@
+
+using System.Collections.Generic;
+
+namespace ImageMagick.Formats;
+
+/// <summary>
+/// Class for defines that are used when a <see cref="MagickFormat.Png"/> image is written.
+/// </summary>
+public sealed class PngWriteDefines : IWriteDefines
+{
+    /// <summary>
+    /// Gets or sets the bit depth for the PNG image.
+    /// Valid values: 1, 2, 4, 8, 16.
+    /// </summary>
+    public uint? BitDepth { get; set; }
+
+    /// <summary>
+    /// Gets or sets the color type of the image.
+    /// </summary>
+    public ColorType? ColorType { get; set; }
+
+    /// <summary>
+    /// Gets or sets the compression filter for the PNG image.
+    /// For compression level 0 (quality value less than 10), the Huffman-only strategy is used, which is fastest but not necessarily the worst compression.
+    /// </summary>
+    public PngCompressionFilter? CompressionFilter { get; set; }
+
+    /// <summary>
+    /// Gets or sets the compression level for the PNG image.
+    /// The compression level ranges from 0 to 9, where 0 indicates no compression and 9 indicates maximum compression.
+    /// For compression level 0 (quality value less than 10), the Huffman-only strategy is used, which is the fastest but not necessarily the worst compression.
+    /// </summary>
+    public uint? CompressionLevel { get; set; }
+
+    /// <summary>
+    /// Gets or sets the compression strategy for the PNG image.
+    /// </summary>
+    public PngCompressionStrategy? CompressionStrategy { get; set; }
+
+    /// <summary>
+    /// Gets or sets the chunks to be excluded.
+    /// </summary>
+    public PngChunkFlags? ExcludeChunks { get; set; }
+
+    /// <summary>
+    /// Gets or sets the chunks to be included.
+    /// </summary>
+    public PngChunkFlags? IncludeChunks { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the PNG decoder should ignore the CRC when writing the image.
+    /// </summary>
+    public bool IgnoreCrc { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the iCCP chunk should be preserved when writing the image.
+    /// </summary>
+    public bool PreserveiCCP { get; set; }
+
+    /// <summary>
+    ///  Gets or sets a value indicating whether ColorMap should be preserve when writing the image.
+    /// </summary>
+    public bool PreserveColorMap { get; set; }
+
+    /// <summary>
+    /// Gets the format where the defines are for.
+    /// </summary>
+    public MagickFormat Format
+        => MagickFormat.Png;
+
+    /// <summary>
+    /// Gets the defines that should be set as a define on an image.
+    /// </summary>
+    public IEnumerable<IDefine> Defines
+    {
+        get
+        {
+            if (BitDepth.HasValue)
+                yield return new MagickDefine(Format, "bit-depth", BitDepth.Value);
+
+            if (ColorType.HasValue)
+                yield return new MagickDefine(Format, "color-type", ColorType.Value);
```

Comment:
This should be a numeric value and not the name of the color type:

```
%               png:color-type can be 0, 2, 3, 4, or 6.
%
%               When png:color-type is 0 (Grayscale), png:bit-depth can
%               be 1, 2, 4, 8, or 16.
%
%               When png:color-type is 2 (RGB), png:bit-depth can
%               be 8 or 16.
%
%               When png:color-type is 3 (Indexed), png:bit-depth can
%               be 1, 2, 4, or 8.  This refers to the number of bits
%               used to store the index.  The color samples always have
%               bit-depth 8 in indexed PNG files.
%
%               When png:color-type is 4 (Gray-Matte) or 6 (RGB-Matte),
%               png:bit-depth can be 8 or 16.
```

This also means we probably need a new enumeration for that.

Reply:
I have created a new PngColorType enum with values for Grayscale, RGB, Indexed, GrayMatte, and RGBMatte. Additionally, I implemented a private method GetPngColorTypeValue to convert these values appropriately, as well as a private method ValidateBitDepth to ensure the bit depth is valid based on the color type.

Here are the key changes:

Enum Creation: Introduced PngColorType enum for clarity and type safety.
Conversion Method: Added GetPngColorTypeValue to map the PngColorType enum to the correct numeric values.
Validation Method: Implemented ValidateBitDepth to check the bit depth against the color type, throwing an ArgumentException for invalid combinations.

I will commit these changes. If any modifications are needed, please let me know. I'm more than happy to make any necessary corrections.

Reply:
I don't think we need the complexity to check for invalid combinations. This might change in ImageMagick in the future. We should leave this up to ImageMagick.

Reply:
Thank you for your feedback. I've removed the validation as suggested.